test/puppetlabs/puppetdb/testutils.clj

```diff
@@ -440,3 +441,31 @@
         table]
        jdbc/query-to-vec
        (map :indexdef)))
+
+(defn update-timestamp [resource resource-list time]
+  (let [current-time  (time/to-date-time time)
+        index (.indexOf resource-list resource)
```

Comment:
This can likely be written using the `map-indexed` func. Might also be useful to mention in the comment below that we can't have identical timestamps due to the sorting which is done in the [testutils/reports/munge-resource-events](https://github.com/puppetlabs/puppetdb/blob/6.x/test/puppetlabs/puppetdb/testutils/reports.clj#L149) function.